lib/grammars/java.js

```diff
@@ -8,7 +8,7 @@ function JavaArgs(filepath, context) {
   const className = GrammarUtils.Java.getClassName(context)
   const classPackages = GrammarUtils.Java.getClassPackage(context)
   const tempFolder = GrammarUtils.createTempFolder("jar-")
-  const cmd = `javac -encoding UTF-8 -sourcepath '${sourcePath}' -d '${tempFolder}' '${filepath}' && java -D'file.encoding'='UTF-8' -cp '${tempFolder}' ${classPackages}${className}`
+  const cmd = `javac -J-Dfile.encoding=UTF-8 -Xlint -sourcepath '${sourcePath}' -d '${tempFolder}' '${filepath}' && java -Dfile.encoding=UTF-8 -cp '${tempFolder}' ${classPackages}${className}`
```

Comment:
Based on the docs `-encoding UTF-8` is a valid way to set the encoding.
https://docs.oracle.com/en/java/javase/13/docs/specs/man/javac.html

What version of Java are you trying this on? I added a test and tested this on Windows, and I had no issues.

```suggestion
  const cmd = `javac -encoding UTF-8 -Xlint -sourcepath '${sourcePath}' -d '${tempFolder}' '${filepath}' && java -Dfile.encoding=UTF-8 -cp '${tempFolder}' ${classPackages}${className}`
```

Reply:
"-encoding UTF-8" will only set source file encoding to UTF-8. Since javac.exe will produce its warning/error messages in its default encoding (for exapmle, in Japanese Windows, it's CP932/ShiftJIS) and in local language (such as Japanese) so it will become garbled. -J-Dfile.encoding=UTF-8 will switch whole output messages of javac.exe to UTF-8.

For example, if you implement java.io.Serializable in your main class and won't set serialversionUID, javac will output warning message, and with '-J-Dfile.encoding=UTF-8', it looks like this:
![image](https://user-images.githubusercontent.com/4508297/118087600-c00cdc00-b400-11eb-876c-6e1ee46b69f2.png)
while only with "-encoding UTF-8" option, this would show:
![image](https://user-images.githubusercontent.com/4508297/118087743-02ceb400-b401-11eb-8df5-75f88d3334a5.png)

for -J option, please check "Standard Options" section in https://docs.oracle.com/en/java/javase/11/tools/javac.html

Reply:
Shouldn't we set both?

Reply:
not necessarily I suppose. -J-Dfile.encoding=UTF-8 will do the job, and I am sure in my environment '-encoding UTF-8' is not there and the package works fine.

Reply:
I added it just in case. Could you update your branch and try it now so I can merge this?